inc/Encryptor.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * WordPress-safe encryption utilities.
+ *
+ * @package OneDesign
+ */
+
+declare( strict_types = 1 );
+
+namespace OneDesign;
+
+/**
+ * Class - Encryptor
+ */
+final class Encryptor {
+	/**
+	 * The salt to use.
+	 *
+	 * @var string
+	 */
+	private static string $salt = 'secure_auth';
+
+	/**
+	 * Length of the initialization vector when using AES-GCM.
+	 *
+	 * @var int
+	 */
+	private const IV_LENGTH = 12;
+
+	/**
+	 * Length of the authentication tag generated by AES-GCM.
+	 *
+	 * @var int
+	 */
+	private const TAG_LENGTH = 16;
+
+	/**
+	 * Encrypts a value using WordPress's built-in encryption.
+	 *
+	 * @param string $value The value to encrypt.
+	 *
+	 * @return string The encrypted value.
+	 *
+	 * @throws \RuntimeException When dependencies are missing or encryption fails.
+	 */
+	public static function encrypt( string $value ): string {
+		$key = self::derive_key();
+		$iv  = random_bytes( self::IV_LENGTH );
+
+		$tag        = '';
+		$ciphertext = openssl_encrypt( $value, 'aes-256-gcm', $key, OPENSSL_RAW_DATA, $iv, $tag );
+		if ( false === $ciphertext ) {
+			throw new \RuntimeException( esc_html__( 'Failed to encrypt value.', 'onedesign' ) );
```

Comment:
My only argument against translating these is if we need to encrpyt/decrpyt something before `plugins_loaded()` - e.g. a `get_option()`. Not sure if that's a real concern in our context.